### ISSUE TITLE
qvm-tags not included in Qubes backups and not re-applied when restoring

### ISSUE THREAD

**Issue by toxub23** [QubesOS/qubes-issues#4167]
### Qubes OS version:
R4.0

### Affected component(s): 
Whonix templates and qubes.UpdatesProxy

### Actual behavior:
I have a Whonix template restored from a backup from Qubes R4. I have the latest Qubes OS R4. I have set whonix-gw / ws -> sys-whonix-s -> sys-firewall-s -> sys-net-s.

In /etc/qubes-rpc/policy/qubes.UpdatesProxy, I have a set of default values for ordinary templates, and among others
$tag:whonix-updatevm $default allow,target=sys-whonix-s
$tag:whonix-updatevm $anyvm deny

Anyway, when I try to update whonix-ws / gw, whonix shows me the information that qubes proxy is not set.
Maybe for the correct operation of whonix I need to edit some additional files?
In qvm-tags whonix-gw / ws, I have "whonix-updatevm".

**Comment by marmarek** [comment]
Does `sys-whonix-s` work (connect to tor etc)?

**Comment by toxub23** [comment]
Yes "sys-whonix-s" works fine. I started "sudo apt-get update" in "sys-whonix-s" for the test and it worked without a problem. However, in "whonix-ws/gw" I still can not get the internet and update the template. When I try to update whonix-ws / gw it showing that I have not configured qubes.UpdatesProxy

**Comment by toxub23** [comment]
Any idea?

**Comment by toxub23** [comment]
Any solutions?

**Comment by adrelanos** [comment]
Install / upgrade to Whonix 14 yet?

**Comment by toxub23** [comment]
Long ago I updated whonix to version 14, before the problem appeared. Maybe this is a backup problem? Maybe when installing whonix from the repository is created in the system not only an entry in updatesProxy? When restoring vm from backup and adding an entry only in updatesProxy, I did not modify anything else and maybe there is a problem here?

**Comment by adrelanos** [comment]
When restoring a backup, I doubt salt settings are restored? I guess re-running

    sudo qubesctl state.sls qvm.anon-whonix

might fix this?

How should backup restoration interact with with salt settings?

**Comment by andrewdavidwong** [comment]
> When restoring a backup, I doubt salt settings are restored? I guess re-running
> 
> ```
> sudo qubesctl state.sls qvm.anon-whonix
> ```
> 
> might fix this?
> 
> How should backup restoration interact with with salt settings?

If a user restores a custom modified Whonix VM from a backup, we shouldn't automatically overwrite the user's custom modifications with the default salt config, since the user may want to keep those custom modifications. (Such overwriting could, however, be an option during the restore process.)

If the original bug report was due to user error (i.e., user restored a custom modified Whonix VM and erroneously expected it to exhibit default behavior), then this issue can be closed as invalid on that basis.

**Comment by adrelanos** [comment]
qvm-tags are not well visible. Only using `qvm-tags` on the dom0 command line.

I want to see that user that will know/remember to restore qvm-tags after VM restoration from backup.

The proper bug fix would seem to be:

> include qvm-tags in Qubes Backups and Restoration

?

Please re-title.

**Comment by andrewdavidwong** [comment]
After re-reading this issue, it is not obvious to me that the problem must lie with VM tags not being restored from backups. However, I will go along with your suggestion, since we can always update further.

**Comment by adrelanos** [comment]
Affects 4.2
